src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamespaceSymbol.cs

```diff
@@ -63,20 +68,39 @@ public sealed override ImmutableArray<Symbol> GetMembers()
         {
             EnsureAllMembersLoaded();
 
-            var memberTypes = GetMemberTypesPrivate();
-
-            if (lazyNamespaces.Count == 0)
-                return StaticCast<Symbol>.From(memberTypes);
+            var memberNamespacesAndTypes = GetMemberNamespacesAndTypesPrivate();
 
-            var builder = ArrayBuilder<Symbol>.GetInstance(memberTypes.Length + lazyNamespaces.Count);
+            return StaticCast<Symbol>.From(memberNamespacesAndTypes);
+        }
 
-            builder.AddRange(memberTypes);
-            foreach (var pair in lazyNamespaces)
+        private ImmutableArray<NamespaceOrTypeSymbol> GetMemberNamespacesAndTypesPrivate()
+        {
+            if (_lazyFlattenedNamespacesAndTypes.IsDefault)
             {
-                builder.Add(pair.Value);
+                if (lazyNamespaces.Count == 0)
+                {
+                    var flattenedTypes = StaticCast<NamespaceOrTypeSymbol>.From(GetMemberTypesPrivate());
+                    ImmutableInterlocked.InterlockedExchange(ref _lazyFlattenedNamespacesAndTypes, flattenedTypes);
+                }
+                else
+                {
+                    ArrayBuilder<NamespaceOrTypeSymbol> builder = ArrayBuilder<NamespaceOrTypeSymbol>.GetInstance();
```

Comment:
can we hint the size here?

Reply:
I could, but to be accurate, it would require enumerating over lazyTypes to get the accumulated length. Let me know if you would like to see that change.

Reply:
ooh, or I could just ensure lazyFlattenedNames is populated first, and use it's length.

Reply:
went ahead and did this